app/Domain/Integrations/Models/IntegrationModel.php

```diff
@@ -153,6 +158,25 @@ public function block(): void
         IntegrationBlocked::dispatch(Uuid::fromString($this->id));
     }
 
+    public function unblock(): void
```

Comment:
I think this will also work

```
public function unblock(): void
{
    /** @var ?IntegrationPreviousStatusModel $integrationPreviousStatus */
    $integrationPreviousStatus = IntegrationPreviousStatusModel::query()->find($this->id);

    $this->update([
        'status' => $integrationPreviousStatus ? $integrationPreviousStatus->status: IntegrationStatus::Draft,
    ]);

    $integrationPreviousStatus?->delete();
}
```